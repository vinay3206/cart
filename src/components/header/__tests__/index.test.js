import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import Header from '../index';

it('renders correctly with Cart link', () => {
  const component = <Header
    showCartLink
    showHomeLink={false}
  />;
  const tree = renderer
    .create(
      <MemoryRouter>{component}</MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with Home link', () => {
  const component = <Header
      showCartLink={false}
      showHomeLink
    />;
  const tree = renderer
    .create(
      <MemoryRouter>{component}</MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with Home and cart link', () => {
  const component = <Header
      showCartLink
      showHomeLink
    />;
  const tree = renderer
    .create(
      <MemoryRouter>{component}</MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with cart link and count', () => {
  const component = <Header
    showCartLink
    showHomeLink={false}
    cartItemsCount={4}
  />;
  const tree = renderer
    .create(
      <MemoryRouter>{component}</MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with home,cart link and count', () => {
  const component = <Header
    showCartLink
    showHomeLink
    cartItemsCount={4}
  />;
  const tree = renderer
    .create(
      <MemoryRouter>{component}</MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
