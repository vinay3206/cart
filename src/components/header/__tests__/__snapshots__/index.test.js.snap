// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with Cart link 1`] = `
<div
  className="header"
>
  <div
    className="pull-right"
  >
    <a
      className="btn-flat btn-red"
      href="/cart"
      onClick={[Function]}
    >
      Cart
      
    </a>
  </div>
</div>
`;

exports[`renders correctly with Home and cart link 1`] = `
<div
  className="header"
>
  <div
    className="pull-right"
  >
    <a
      className="btn-flat btn-blue-light"
      href="/home"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      className="btn-flat btn-red"
      href="/cart"
      onClick={[Function]}
    >
      Cart
      
    </a>
  </div>
</div>
`;

exports[`renders correctly with Home link 1`] = `
<div
  className="header"
>
  <div
    className="pull-right"
  >
    <a
      className="btn-flat btn-blue-light"
      href="/home"
      onClick={[Function]}
    >
      Home
    </a>
  </div>
</div>
`;

exports[`renders correctly with cart link and count 1`] = `
<div
  className="header"
>
  <div
    className="pull-right"
  >
    <a
      className="btn-flat btn-red"
      href="/cart"
      onClick={[Function]}
    >
      Cart
       (4)
    </a>
  </div>
</div>
`;

exports[`renders correctly with home,cart link and count 1`] = `
<div
  className="header"
>
  <div
    className="pull-right"
  >
    <a
      className="btn-flat btn-blue-light"
      href="/home"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      className="btn-flat btn-red"
      href="/cart"
      onClick={[Function]}
    >
      Cart
       (4)
    </a>
  </div>
</div>
`;
